packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -187,6 +192,8 @@ export interface EngineReportingOptions<TContext> {
   generateClientInfo?: GenerateClientInfo<TContext>;
 }
 
+const validationErrorStaticIdentifier = "GraphQLValidationFailure";
```

Comment:
It looks like you are using this twice, as both query signature and operation name?

Instead of going out of your way to make this special identifier look exactly like normal stats record keys, why not actually make it differentiable?  Let's keep it starting with `#` to preserve the formerly-not-quite-but-now-actually invariant that stats report keys are parseable, but not `# ` like `fun signatureAndOperation` looks for.

So eg, let's have the entire statsReportKey be `## GraphQLParseFailure` or `## GraphQLValidationFailure`.

Reply:
Why the double `#`?

Reply:
To make it distinct from actual signatures.

Reply:
good idea! Done.